src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -411,18 +411,94 @@ void flag_temporal_id_as_pending(const gsl::not_null<db::DataBox<DbTags>*> box,
   // temporal_ids that we have already completed interpolation on.  So
   // here we do not add any temporal_ids that are already present in
   // `ids` or `completed_ids`.
-  db::mutate_apply<tmpl::list<Tags::PendingTemporalIds<TemporalId>>,
+  db::mutate_apply<tmpl::list<Tags::PendingTemporalIds<TemporalId>,
+                              Tags::PendingPendingTemporalIds<TemporalId>>,
                    tmpl::list<Tags::TemporalIds<TemporalId>,
                               Tags::CompletedTemporalIds<TemporalId>>>(
-      [&temporal_id](const gsl::not_null<std::deque<TemporalId>*> pending_ids,
-                     const std::deque<TemporalId>& ids,
-                     const std::deque<TemporalId>& completed_ids) {
+      [&temporal_id](
+          const gsl::not_null<std::deque<TemporalId>*> pending_ids,
+          const gsl::not_null<std::deque<TemporalId>*> pending_pending_ids,
+          const std::deque<TemporalId>& ids,
+          const std::deque<TemporalId>& completed_ids) {
         if (std::find(completed_ids.begin(), completed_ids.end(),
                       temporal_id) == completed_ids.end() and
             std::find(ids.begin(), ids.end(), temporal_id) == ids.end() and
             std::find(pending_ids->begin(), pending_ids->end(), temporal_id) ==
                 pending_ids->end()) {
-          pending_ids->push_back(temporal_id);
+          if constexpr (std::is_same_v<TemporalId, LinkedMessageId<double>>) {
+            if (alg::found(*pending_pending_ids, temporal_id)) {
+              // Do nothing
+              return;
+            } else {
+              const auto print_pending = [&](const std::string& pending_str) {
+                Parallel::printf(
+                    "%s: Adding %s to %s\n"
+                    " Pending: %s\n"
+                    " Pending pending: %s\n",
+                    pretty_type::name<InterpolationTargetTag>(), temporal_id,
+                    pending_str, *pending_ids, *pending_pending_ids);
+              };
+              const auto check_and_place_id =
+                  [&](const std::deque<TemporalId>& ids_to_check) {
+                    ASSERT(temporal_id.previous.has_value(),
+                           "If checking if this LinkedMessageId is in order, "
+                           "then this can't be the "
+                           "first id.");
+                    if (temporal_id.previous.value() ==
+                        ids_to_check.back().id) {
+                      pending_ids->push_back(temporal_id);
+                      print_pending("pending (pending not empty)");
+                    } else {
+                      pending_pending_ids->push_back(temporal_id);
+                      print_pending("pending pending (pending not empty)");
+                    }
+                  };
+              // Cases: (1p = pending, 2p = pending pending)
```

Comment:
Hmm possibly. I think in order to make this change, the bulk of this logic would move into `VerifyTemporalIdsAndSendPoints` because there it loops over pending ids to see if they are ready, and then this function would basically remain unchanged (except now the sort of pending ids would always be needed to ensure the earliest id is first).

I also think with all the changes in this PR that there can only be a single id in `Tags::TemporalIds` at a time (since only one horizon find can be happening at once). That makes more things easier I think.

Reply:
If you're not familiar with the [priority queue](https://en.cppreference.com/w/cpp/container/priority_queue) I suggest you read up on that data structure.

Reply:
I looked over the priority queue and found that it won't be suitable for our case. Within in the source code (and tests) we need to search through the queue to see if an id is already there and the priority queue does not offer iterators or really any way to query what is in the queue besides the `top` value and size. Therefore, I'm just going to stick with a `deque` and sort it.